fdbclient/include/fdbclient/KeyBackedTypes.actor.h

```diff
@@ -0,0 +1,1083 @@
+			if (!rangeResult.results.empty()) {
+				// kvs.more is known to be true and kvs is not empty since the filtered result set is not empty.  Set
+				// the output rangeResult.more to true if the last raw result was within the map range, else false.
+				rangeResult.more = self.subspace.contains(kvs.back().key);
+				break;
+			}
+
+			// At this point, the filtered rangeResult is empty but there may be more raw results in the db.  We cannot
+			// return this rangeResult to the caller because the caller won't know which key to start reading at for the
+			// next chunk, so we will use the raw keys to read the next chunk here and repeat the loop.
+			if (reverse) {
+				// The last key is the end of the new getRange, which includes the back key with orEqual because
+				// getRange end is exclusive
+				ksEnd = ::firstGreaterOrEqual(kvs.back().key);
+			} else {
+				// The last key is the exclusive begin of the new getRange
+				ksBegin = ::firstGreaterThan(kvs.back().key);
+			}
+			getRangeFuture = tr->getRange(ksBegin, ksEnd, limits, snapshot, reverse);
+		}
+
+		return rangeResult;
+	}
+
+	// GetRange with typed KeySelectors
+	template <class Transaction>
+	Future<RangeResultType> getRange(Transaction tr,
+	                                 KeySelector begin,
+	                                 KeySelector end,
+	                                 GetRangeLimits limits,
+	                                 Snapshot snapshot = Snapshot::False,
+	                                 Reverse reverse = Reverse::False) const {
+		return getRangeActor(*this, tr, begin, end, limits, snapshot, reverse);
+	}
+
+	// Find the closest key which is <, <=, >, or >= query
+	// This is like resolving a KeySelector to a key but it will return not-present if the key is outside of the map
+	// subspace. It does this without visiting any range outside of the map subspace, so it succeeds if the map is
+	// located next to a shard which is currently unavailable or, more likely, an adjacent key has been modified by
+	// a VersionStamp atomic op so it is not allowed to be read in a range read operation.
+	ACTOR template <class Transaction>
+	static Future<Optional<KVType>> seek(KeyBackedMap self,
+	                                     Transaction tr,
+	                                     KeyType query,
+	                                     bool lessThan,
+	                                     bool orEqual,
+	                                     Snapshot snapshot) {
+		// Operations map to the following getRange operations
+		// <  query  getRange begin               query   1                reverse
+		// <= query  getRange begin               keyAfter(query)          reverse
+		// >= query  getRange key                 end   1                  forward
+		// >  query  getRange keyAfter(query)     end   1                  forward
+		Key begin;
+		Key end;
+
+		if (lessThan) {
+			begin = self.subspace.begin;
+			end = self.packKey(query);
+			if (orEqual) {
```

Comment:
I think this is a different definition of `orEqual` than key selectors. I'm ok with that, but we may want to comment on that

Reply:
This comment
```
	// Find the closest key which is <, <=, >, or >= query
```
so perhaps I should just remove the comparison to KeySelectors in the comment to avoid confusion?  KeySelectors are not actually involved in the method, this is really meant to just implement the common container edge-seek concepts.

Reply:
I updated the comments for these methods to be more clear.